lib/routes/charge/pos_invoice.dart

```diff
@@ -72,11 +73,22 @@ class POSInvoiceState extends State<POSInvoice> with TickerProviderStateMixin {
     if (accountSubscription == null) {
       AccountBloc accountBloc = AppBlocsProvider.of<AccountBloc>(context);
       accountSubscription = accountBloc.accountStream.listen((acc) {
-        currentCurrency = _useFiat
-            ? CurrencyWrapper.fromFiat(acc.fiatCurrency)
-            : CurrencyWrapper.fromBTC(acc.currency);
+        currentCurrency = acc.posCurrency;
+      });
+      FetchRates fetchRatesAction = FetchRates();
+      accountBloc.userActionsSink.add(fetchRatesAction);
+      fetchRatesAction.future.catchError((err) {
+        if (this.mounted) {
+          UserProfileBloc userProfileBloc =
+              AppBlocsProvider.of<UserProfileBloc>(context);
+          setState(() {
+            userProfileBloc.posCurrencySink.add("BTC");
```

Comment:
Why do we need this? We don't need to manipulate the user choice. If we failed to retrieve the rates, let's just render the default (BTC? Sats?).

Reply:
[Done](https://github.com/breez/breezmobile/pull/309/commits/2634aa19b1891d79709f56eaff5292aeea591d8f)